src/EbmlFloat.cpp

```diff
@@ -113,26 +113,31 @@ uint64 EbmlFloat::UpdateSize(bool bWithDefault, bool  /* bForceRender */)
 filepos_t EbmlFloat::ReadData(IOCallback & input, ScopeMode ReadFully)
 {
   if (ReadFully != SCOPE_NO_DATA) {
-    binary Buffer[20];
-    assert(GetSize() <= 20);
-    input.readFully(Buffer, GetSize());
-
-    if (GetSize() == 4) {
-      big_int32 TmpRead;
-      TmpRead.Eval(Buffer);
-      auto tmpp = int32(TmpRead);
-      float val;
-      memcpy(&val, &tmpp, 4);
-      Value = static_cast<double>(val);
-      SetValueIsSet();
-    } else if (GetSize() == 8) {
-      big_int64 TmpRead;
-      TmpRead.Eval(Buffer);
-      auto tmpp = int64(TmpRead);
-      double val;
-      memcpy(&val, &tmpp, 8);
-      Value = val;
-      SetValueIsSet();
+    assert(GetSize() == 4 || GetSize() == 8);
+    if (GetSize() == 4 || GetSize() == 8) {
```

Comment:
For improved readability I'd prefer to short-circuit on the error case:

```
  assert(…);
  if (GetSize() != 4 && GetSize() != 8) {
    input.setFilePointer(GetSize(), seek_current);
    return GetSize();
  }
```

It would also save one level of indentation.